### PR TITLE
Bug 1827166: pkg/cvo/sync_worker: Log precondition handling

### PR DESCRIPTION
**Pull request by wking**
To make it easier to debug precondition failures like [rhbz#1827166][1].

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1827166

### CODE REVIEW

**Comment by wking** [comment]
4.4 backport created manually in #361.

**Comment by wking** [comment]
[e2e-aws-upgrade][1]:

```
level=info msg="Cluster operator authentication Progressing is True with _WellKnownNotReady: Progressing: got '404 Not Found' status while trying to GET the OAuth well-known https://10.0.128.43:6443/.well-known/oauth-authorization-server endpoint data"
level=info msg="Cluster operator authentication Available is False with : "
level=info msg="Cluster operator cloud-credential Progressing is True with Reconciling: 3 of 4 credentials requests provisioned, 0 reporting errors."
level=info msg="Cluster operator console Progressing is True with SyncLoopRefresh_InProgress: SyncLoopRefreshProgressing: Working toward version 0.0.1-2020-04-24-202827"
level=info msg="Cluster operator console Available is False with Deployment_InsufficientReplicas: DeploymentAvailable: 0 pods available for console deployment"
level=error msg="Cluster operator etcd Degraded is True with NodeController_MasterNodesReady: NodeControllerDegraded: The master nodes not ready: node \"ip-10-0-143-80.us-west-2.compute.internal\" not ready since 2020-04-24 20:48:57 +0000 UTC because NodeStatusUnknown (Kubelet stopped posting node status.)"
level=info msg="Cluster operator insights Disabled is False with : "
level=error msg="Cluster operator kube-apiserver Degraded is True with NodeController_MasterNodesReady: NodeControllerDegraded: The master nodes not ready: node \"ip-10-0-143-80.us-west-2.compute.internal\" not ready since 2020-04-24 20:48:57 +0000 UTC because NodeStatusUnknown (Kubelet stopped posting node status.)"
level=info msg="Cluster operator kube-apiserver Progressing is True with NodeInstaller: NodeInstallerProgressing: 3 nodes are at revision 2; 0 nodes have achieved new revision 6"
level=error msg="Cluster operator kube-controller-manager Degraded is True with NodeController_MasterNodesReady::NodeInstaller_InstallerPodFailed: NodeControllerDegraded: The master nodes not ready: node \"ip-10-0-143-80.us-west-2.compute.internal\" not ready since 2020-04-24 20:48:57 +0000 UTC because NodeStatusUnknown (Kubelet stopped posting node status.)\nNodeInstallerDegraded: 1 nodes are failing on revision 5:\nNodeInstallerDegraded: static pod of revision 5 has been installed, but is not ready while new revision 6 is pending"
level=info msg="Cluster operator kube-controller-manager Progressing is True with NodeInstaller: NodeInstallerProgressing: 1 nodes are at revision 0; 2 nodes are at revision 4; 0 nodes have achieved new revision 7"
level=error msg="Cluster operator kube-scheduler Degraded is True with NodeController_MasterNodesReady: NodeControllerDegraded: The master nodes not ready: node \"ip-10-0-143-80.us-west-2.compute.internal\" not ready since 2020-04-24 20:48:57 +0000 UTC because NodeStatusUnknown (Kubelet stopped posting node status.)"
level=info msg="Cluster operator kube-scheduler Progressing is True with NodeInstaller: NodeInstallerProgressing: 2 nodes are at revision 5; 1 nodes are at revision 6"
level=error msg="Cluster operator machine-config Degraded is True with RequiredPoolsFailed: Failed to resync 0.0.1-2020-04-24-202827 because: timed out waiting for the condition during syncRequiredMachineConfigPools: error pool master is not ready, retrying. Status: (pool degraded: false total: 3, ready 2, updated: 3, unavailable: 1)"
level=info msg="Cluster operator machine-config Available is False with : Cluster not available for 0.0.1-2020-04-24-202827"
level=info msg="Cluster operator monitoring Available is False with : "
level=info msg="Cluster operator monitoring Progressing is True with RollOutInProgress: Rolling out the stack."
level=error msg="Cluster operator monitoring Degraded is True with UpdatingnodeExporterFailed: Failed to rollout the stack. Error: running task Updating node-exporter failed: reconciling node-exporter DaemonSet failed: updating DaemonSet object failed: waiting for DaemonSetRollout of node-exporter: daemonset node-exporter is not ready. status: (desired: 6, updated: 6, ready: 5, unavailable: 1)"
level=error msg="Cluster operator network Degraded is True with RolloutHung: DaemonSet \"openshift-multus/multus\" rollout is not making progress - last change 2020-04-24T20:53:49Z\nDaemonSet \"openshift-sdn/ovs\" rollout is not making progress - last change 2020-04-24T20:53:41Z\nDaemonSet \"openshift-sdn/sdn\" rollout is not making progress - last change 2020-04-24T20:53:27Z"
level=info msg="Cluster operator network Progressing is True with Deploying: DaemonSet \"openshift-multus/multus\" is not available (awaiting 1 nodes)\nDaemonSet \"openshift-sdn/ovs\" is not available (awaiting 1 nodes)\nDaemonSet \"openshift-sdn/sdn\" is not available (awaiting 1 nodes)"
level=error msg="Cluster operator openshift-apiserver Degraded is True with APIServerDeployment_UnavailablePod: APIServerDeploymentDegraded: 1 of 3 requested instances are unavailable for apiserver.openshift-apiserver"
level=info msg="Cluster operator openshift-controller-manager Progressing is True with _DesiredStateNotYetAchieved: Progressing: daemonset/controller-manager: updated number scheduled is 2, desired number scheduled is 3"
level=fatal msg="failed to initialize the cluster: Cluster operator machine-config is reporting a failure: Failed to resync 0.0.1-2020-04-24-202827 because: timed out waiting for the condition during syncRequiredMachineConfigPools: error pool master is not ready, retrying. Status: (pool degraded: false total: 3, ready 2, updated: 3, unavailable: 1)" 
```

Oh, man.

/retest

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/360/pull-ci-openshift-cluster-version-operator-master-e2e-aws-upgrade/608

File: pkg/cvo/sync_worker.go [hunk 1]
```diff
@@ -512,7 +512,11 @@ func (w *SyncWorker) syncOnce(ctx context.Context, work *SyncWork, maxWorkers in
 		payloadUpdate.LoadedAt = time.Now()
 
 		// need to make sure the payload is only set when the preconditions have been successful
-		if !info.Local && len(w.preconditions) > 0 {
+		if len(w.preconditions) == 0 {
+			klog.V(4).Info("No preconditions configured.")
```
Comment (LalatenduMohanty) [inline pkg/cvo/sync_worker.go:516]: Nit pick `No preconditions found` sounds better?

**Comment by wking** [comment]
Setting `lgtm` myself while we work out @LalatenduMohanty 's creds.

**Pull request merged**